src/audio_unit/render_callback.rs

```diff
@@ -398,7 +401,7 @@ impl AudioUnit {
         // First, we'll retrieve the stream format so that we can ensure that the given callback
         // format matches the audio unit's format.
         let id = sys::kAudioUnitProperty_StreamFormat;
-        let asbd = try!(self.get_property(id, Scope::Output, Element::Output));
+        let asbd = try!(self.get_property(id, Scope::Input, Element::Output));
```

Comment:
As per https://github.com/RustAudio/coreaudio-rs/issues/60 I think Output/Output gives the hardware stream format, not the user requested one.